app/uk/gov/hmrc/trackingconsentfrontend/assets/src/interfaces/optimizelyCommunicatorFactory.ts

```diff
@@ -5,13 +5,37 @@ const optimizelyCommunicatorFactory = (window: Window): Communicator => ({
   sendPreferences: (userPreferences: UserPreferences) => {
 
     const setOptimizelyOptOut = (isOptOut: Boolean) => {
-      window.optimizely.push({ 'type': 'optOut', isOptOut });
+      window.optimizely.push({ 'type': 'optOut', isOptOut })
     }
 
-    window.optimizely = window.optimizely || [];
+    const setupOptimizelyIfConfigured = () => {
+      const regex = /^(\/|(https:)?\/\/www\.tax\.service\.gov\.uk\/|(https:)?\/\/www\.(qa|dev|staging)\.tax\.service\.gov\.uk\/|(http:)?\/\/localhost:12345\/)tracking-consent/
```

Comment:
This regex is something that is highly likely to need to change and require ongoing maintenance. For example, if the platform url structure changes, or the port number changes etc. The length of it also makes it difficult to easily see what it is doing. Additionally, I couldn't see any tests to exercise the different types of expected inputs. To make the code more flexible, I would probably aim to do something simpler like just doing a `.find` to look for the script tag containing the 'data-optimizely-id' attribute i.e. `scriptTags.find(scriptTag => scriptTag.getAttribute('data-optimizely-id') )`